sdks/go/test/integration/xlang/expansion_test.go

```diff
@@ -0,0 +1,51 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xlang
+
+import (
+	"os"
+	"testing"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core/runtime/xlangx/expansionx"
+	"github.com/apache/beam/sdks/v2/go/test/integration"
+)
+
+const (
+	// TODO(BEAN-13505): Select the most recent Beam release instead of a hard-coded
+	// string.
+	beamVersion  = "2.34.0"
+	gradleTarget = ":sdks:java:io:expansion-service:runExpansionService"
+)
+
+func TestAutomatedExpansionService(t *testing.T) {
+	integration.CheckFilters(t)
+	jarPath, err := expansionx.GetBeamJar(gradleTarget, beamVersion)
+	if err != nil {
+		t.Fatalf("failed to get JAR path, got %v", err)
+	}
+	t.Cleanup(func() { os.Remove(jarPath) })
+
+	serviceRunner := expansionx.NewExpansionServiceRunner(jarPath, "8097")
+	err = serviceRunner.StartService()
+	if err != nil {
+		t.Errorf("failed to start expansion service JAR, got %v", err)
+	}
+
```

Comment:
My only concern is whether we should send a ping or *something* to actually validate that the jar is running a service at the port. Right now, black-box, the service runner could be replaced with no-ops and pass.

Otherwise it LGTM.

Reply:
Fair point, I will play around with finding a way to check that the service is what we expect.

Reply:
Added a ping to the local port to make sure we could connect. As a consequence I discovered that a brief sleep to let the JAR start up was necessary (cmd.Start() is non-blocking in starting up its own process.)

Reply:
Makes sense. a 2 second wait isn't ideal, but I can't think of a clean way around things and do them asynchronously, so a wait we shall have.